### PR TITLE
Add MedicalNoteList, hook MedicalNote class into program

### PR DESCRIPTION
**Pull request by wang-jun-hao**
Implement parser support (NoteCommand, NoteCommandParser)

### CODE REVIEW

File: src/test/java/seedu/medibook/logic/commands/NoteCommandTest.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package seedu.medibook.logic.commands;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class NoteCommandTest {
+
+}
```
Comment (PrestonTYR) [inline src/test/java/seedu/medibook/logic/commands/NoteCommandTest.java]: Did you forget to add the test cases? 😄

File: src/main/java/seedu/medibook/logic/parser/MediBookParser.java [hunk 2]
```diff
@@ -68,6 +77,12 @@ public Command parseCommand(String userInput) throws ParseException {
         case HelpCommand.COMMAND_WORD:
             return new HelpCommand();
 
+        case NoteCommand.COMMAND_WORD:
+            return new NoteCommandParser(new Patient(new Ic("S9806137A"), new Name("junhao"),
+                    new DateOfBirth("19-02-1998"), new Phone("90068112"),
+                    Optional.empty(), Optional.empty(),
+                    Optional.empty(), Optional.empty(), Optional.empty(), new HashSet<Tag>())).parse(arguments);
+
```
Comment (PrestonTYR) [inline src/main/java/seedu/medibook/logic/parser/MediBookParser.java]: Should the NoteCommandParser constructor be taking a preset patient as a field?

File: src/test/java/seedu/medibook/logic/commands/NoteCommandTest.java (hunk 1 above)
Comment (wang-jun-hao) [reply src/test/java/seedu/medibook/logic/commands/NoteCommandTest.java]: Will do integration testing after other modules are merged within this branch

**Review by PrestonTYR — APPROVED** [review]
lgtm

**Pull request merged**